proposals/0327-actor-initializers.md

```diff
@@ -42,126 +49,129 @@ The protection is achieved by _isolating_ the mutable state of each actor instan
+class NonSendableAhmed { 
+  var state: Int = 0
 }
 
-actor ConnectionManager {
-  var status: ConnectionStatusDelegate
-  var connectionCount: Int
-
-  @MainActor
-  init(_ sts: ConnectionStatusDelegate) async {
-    // --- on MainActor --
-    self.status = sts
-    self.status.connectionStarting()
-    self.connectionCount = 0
-    // --- self fully-initialized here ---
-    
-    // ... connect ...
-    self.status.connectionEstablished()
+@MainActor
+class Maria {
+  let friend: NonSendableAhmed
+
+  init() {
+    self.friend = NonSendableAhmed()
+  }
+
+  init(sharingFriendOf otherMaria: Maria) {
+    // While the friend is non-Sendable, this initializer and
+    // and the otherMaria are isolated to the MainActor. That is,
+    // they share the same executor. So, it's OK for the non-Sendable value
+    // to cross between otherMaria and self.
+    self.friend = otherMaria.friend
+  }
+
+  deinit {
+    friend.state += 1   // 💥 the deinit is not isolated to the MainActor,
+                        // so this mutation can happen concurrently with other
+                        // accesses to the same underlying instance of 
+                        // NonSendableAhmed.
   }
 }
+
+func example() async {
+  let m1 = await Maria()
+  let m2 = await Maria(sharingFriendOf: m1)
+  doSomething(m1, m2)
+}
 ```
 
-then which executor should be used? Should it be valid to isolate an actor's `init` to a global actor, such as the `@MainActor`, to ensure that the right executor is used for the operations it performs? The example above serves as a possible use case for that capability: being able to perform the initialization while on `@MainActor` so that the `ConnectionStatusDelegate` can be updated without any possibility of suspension (i.e., no `await` needed). 
+In the example above, access to isolated and non-Sendable stored properties of `Maria` from the `deinit` is not safe, as it can introduce a data race. That's because a `deinit` (and its callers) cannot hop to the `MainActor`'s executor in order to guarantee safe access, like an isolated method could. If the `friend` were `Sendable`, then there is no problem.
 
-The existing implementation makes it impossible to write a correct `init` for the example above, because 
-the `init` is considered to be entirely isolated to the `@MainActor`. Thus, it's not possible to initialize `self.status` _at all_. It's not possible to `await` and hop to `self`'s executor to perform an assignment to `self.status`, because `self` is not a fully-initialized actor-instance yet!
+Also, the reference count of `self` upon entering the `deinit` is zero. So, it is undefined behavior for that `self` instance to then escape the `deinit`, living beyond the lifetime of the `deinit`. This is a problem for ordinary classes, too, leading to [random crashes](https://bugs.swift.org/browse/SR-6942). Since lifetime extension issues in an actor's `deinit` is a general problem, we defer a solution to that for a future proposal.
```

Comment:
> So, it is undefined behavior for that `self` instance to then escape the `deinit`, living beyond the lifetime of the `deinit`.

That's the important bit -- up a few lines we say tho that it "keeps it alive" -- should we say always that it is undefined behavior to refer to self after the deinit has completed?

Reply:
resolved with an ❌ emoji on that code comment to keep the messaging consistent, that it's not valid to do that.